### PR TITLE
Webhooks base controller

### PR DESCRIPTION
**Pull request by p55**
solves https://github.com/BookingSync/bookingsync_application/issues/20

### CODE REVIEW

File: app/controllers/bookingsync_application/webhooks/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def verify_signature
+    Rails.logger.debug "bookingsync_hook_raw_post: #{request.raw_post}"
+    unless signature == request.headers["X-Content-Signature"]
+      invalidate_request_and_log "Bad BookingSync signature"
+    end
+  end
+
+  def signature
```
Comment (ZenCocoon) [inline app/controllers/bookingsync_application/webhooks/base_controller.rb]: I think we should directly include support for the new signature process (with strict base64).

**Comment by ZenCocoon** [comment]
This needs to be tested.

File: app/controllers/bookingsync_application/webhooks/base_controller.rb (hunk 1 above)
Comment (Mandaryn) [inline app/controllers/bookingsync_application/webhooks/base_controller.rb]: lets start with the current way, and the other way should be introduced together with pr to core actually fixing this? but his should be tested in the dummy app, simple controller spec with correct and incorrect payload, pick some sample webhook from any app

Comment (ZenCocoon) [inline app/controllers/bookingsync_application/webhooks/base_controller.rb]: I think it's best to have the first version released handling both, much easier to ensure everyone is up to speed before we migrate the signature. It's super simple as well. Just need to use http://ruby-doc.org/stdlib-2.3.0/libdoc/base64/rdoc/Base64.html#method-i-strict_encode64 for the new signature.

:+1: For dummy app testing

File: app/controllers/bookingsync_application/webhooks/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  protected
+
+  def invalidate_request_and_log(message)
+    logger.info "Failing with 422 because: #{message}"
+    head 422
+  end
+
+  def verify_signature
+    Rails.logger.debug "bookingsync_hook_raw_post: #{request.raw_post}"
+    unless signature == request.headers["X-Content-Signature"]
```
Comment (ZenCocoon) [inline app/controllers/bookingsync_application/webhooks/base_controller.rb]: `unless [old_signature, new_signature].include? request.headers["X-Content-Signature"]`

**Comment by ZenCocoon** [comment]
Looking great to me. @Mandaryn wanna review?

**Comment by Mandaryn** [comment]
for me the specs should actually test decoding the payload

**Comment by Mandaryn** [comment]
hooo, maybe i read it all too fast

**Comment by ZenCocoon** [comment]
it does, encrypting "foo" right now

**Comment by Mandaryn** [comment]
ok, looks ok

**Pull request merged**